src/components/appSwitch/AppSwitch.js

```diff
@@ -252,17 +214,134 @@ const AppSwitch = (props: LoadingProps) => {
+            buttons: [
+              {
+                text: 'Cancel',
+                onPress: dismiss => {
+                  res(false)
+                  showDialog({
+                    title: t`Change wallet`,
+                    message: t`Back to ${truncateMiddle(account, 20)}`,
+                    loading: true,
+                    showCloseButtons: false,
+                    buttons: [],
+                  })
+                },
+                mode: 'text',
+                color: 'lighterGray',
+              },
+              {
+                text: 'Logout',
+                color: 'red',
+                onPress: dismiss => {
+                  AsyncStorage.clear()
+                  res(true)
+                  dismiss()
+                },
+              },
+            ],
+          }),
+        )
+        const choice = await dialogPromise
+        log.debug('initWalletAndStorage: account changed , performing login again', { choice })
+        if (choice === false) {
+          return
+        }
+      } else if (initializing.current || web3 === false) {
+        return
+      }
+      try {
+        initializing.current = true
+        log.debug('initWalletAndStorage:', { provider, web3 })
+        let web3Result = web3
+        if (!web3 && 'WEB3WALLET' === provider) {
+          web3Result = await walletConnect().catch()
+          if (!web3Result) {
+            return
+          }
+          log.debug('initWalletAndStorage walletConnect:', { web3Result })
+          if (publicKey && publicKey !== web3Result.eth.defaultAccount) {
+            //if different account then logged in, then wait for the logout dialog
+            initializing.current = false
+            log.debug('initWalletAndStorage walletConnect different account then logged in:', {
+              web3Result,
+              publicKey,
+            })
+
+            return
+          }
+        }
+
+        await initWalletAndStorage(web3Result, provider)
+
+        log.debug('initWalletAndStorage: storage and wallet ready')
+      } catch (e) {
+        if ('UnsignedJWTError' === e.name) {
+          return restartWithMessage(
+            "You haven't used GoodDollar app on this device for a long time. " +
+              'You need to sign in again. Make sure to use the same account you previously signed in with.',
+          )
+        }
+
+        // if error in realmdb logout the user, he needs to signin/signup again
+        log.error('failed initWalletAndStorage', e.message, e)
+
+        if (e.message.includes('realmdb')) {
+          return restartWithMessage(
+            'We are sorry, but due to database upgrade, you need to perform the Signup process again. ' +
+              'Make sure to use the same account you previously signed in with.',
+          )
+        }
+
+        restartWithMessage('Wallet could not be loaded. Please refresh.')
+      }
+    }
+    run()
+  }, [initWalletAndStorage, initializing, web3, goodWallet])
```

Comment:
too large code block, i'd suggest to split it onto few functions:

```suggestion
    const askForWalletChange = () => new Promise(resolve => {
      const [oldAddress, newAddress] = [account, web3.eth.defaultAccount]
        .map(address => truncateMiddle(address, 20))

      const prompt = mustache(
        t`Your wallet address has changed. Do you want to logout from {oldAddress} and login with: {newAddress} ?`,
        { oldAddress, newAddress }
      )

      const buttomPrompt = mustache(
        t`Back to {oldAddress}`,
        { oldAddress }
      )

      showDialog({
        image: <InfoIcon />,
        showCloseButtons: false,
        title: t`Do you want to logout?`,
        message: prompt,
        buttons: [
          {
            text: 'Cancel',
            onPress: dismiss => {
              resolve(false)
              showDialog({
                title: t`Change wallet`,
                message: buttomPrompt,
                loading: true,
                showCloseButtons: false,
                buttons: [],
              })
            },
            mode: 'text',
            color: 'lighterGray',
          },
          {
            text: 'Logout',
            color: 'red',
            onPress: dismiss => {
              AsyncStorage.clear()
              resolve(true)
              dismiss()
            },
          },
        ],
      })
    })

    const walletConnectOrInitialize = async () => {
      let web3Result = web3
      
      initializing.current = true
      log.debug('initWalletAndStorage:', { provider, web3 })

      if (!web3 && 'WEB3WALLET' === provider) {
        web3Result = await walletConnect().catch()

        if (!web3Result) {
          return
        }

        log.debug('initWalletAndStorage walletConnect:', { web3Result })

        if (publicKey && publicKey !== web3Result.eth.defaultAccount) {
          // if different account then logged in, then wait for the logout dialog
          initializing.current = false
          
          log.debug('initWalletAndStorage walletConnect different account then logged in:', {
            web3Result,
            publicKey,
          })

          return
        }
      }

      await initWalletAndStorage(web3Result, provider)
      log.debug('initWalletAndStorage: storage and wallet ready')
    }

    const onInitializationFailed = e => {
      if ('UnsignedJWTError' === e.name) {
        return restartWithMessage(
          "You haven't used GoodDollar app on this device for a long time. " +
            'You need to sign in again. Make sure to use the same account you previously signed in with.',
        )
      }

      // if error in realmdb logout the user, he needs to signin/signup again
      log.error('failed initWalletAndStorage', e.message, e)

      if (e.message.includes('realmdb')) {
        return restartWithMessage(
          'We are sorry, but due to database upgrade, you need to perform the Signup process again. ' +
            'Make sure to use the same account you previously signed in with.',
        )
      }

      restartWithMessage('Wallet could not be loaded. Please refresh.')
    }

    const bootstrap = async () => {
      const provider = await AsyncStorage.getItem(GD_PROVIDER)
      const isThirdParty = provider === 'WEB3WALLET'

      if (isThirdParty === false && initializing.current) {
        // if torus and initializing nothing to do here anymore
        return
      }

      if (web3 === undefined && initializing.current) {
        // in case web3 was disconnected
        initializing.current = undefined
      }

      // if we are already initializing or wallet not ready
      if (initializing.current && web3 && goodWallet && goodWallet.account === web3.eth.defaultAccount) {
        // in case user switched back to his account
        return hideDialog()
      }

      let [publicKey] = (await AsyncStorage.getItem(GD_USER_PRIVATEKEYS)) || []

      if (
        (web3 && publicKey !== web3.eth.defaultAccount) ||
        (web3 && goodWallet && goodWallet.account !== web3.eth.defaultAccount)
      ) {
        const account = publicKey || goodWallet?.account
        const changeConfirmed = await askForWalletChange()
        
        log.debug('initWalletAndStorage: account changed , performing login again', { changeConfirmed })

        if (changeConfirmed === false) {
          return
        }
      } else if (initializing.current || web3 === false) {
        return
      }
      
      await walletConnectOrInitialize().catch(onInitializationFailed)
      initializing.current = false
    }

    log.debug('initwalletandstorage start:', { initializing, web3, goodWallet, connecting })
    bootstrap()
  }, [initWalletAndStorage, initializing, web3, goodWallet])
```

Reply:
also strings to be transllated **shouldn't contain** embedded variables. because the evaluated string will be used as i18n key and it will be different for the different var values.

an constant-only string should be used, with some placeholders to substitute values
we're using an mustache-like templates (implemented via lodash's template() fn)
```javascript
import mustache from 'path/to/lib/utils/mustache'
```